dev-docs/publisher-api-reference.md

```diff
@@ -1290,4 +1285,79 @@ This call returns the following DFP video ad tag URL:
 https://pubads.g.doubleclick.net/gampad/ads?env=vp&gdfp_req=1&output=vast&unviewed_position_start=1&correlator=1507127916397&sz=640x480&url=http://www.referer-url.com&iu=/19968336/prebid_cache_video_adunit&cust_params=hb_bidder%3DappnexusAst%26hb_adid%3D26d4996ee83709%26hb_pb%3D10.00%26hb_size%3D640x480%26hb_uuid%3D16c887cf-9986-4cb2-a02f-8e9bd025f875%26section%3Dblog%26anotherKey%3DanotherValue&hl=en
 ```
 
+In this example, we use the [`options.url`](#buildVideoUrl-options-object) argument to pass in the video ad server directly:
+
+```javascript
+var adserverTag = 'https://pubads.g.doubleclick.net/gampad/ads?'
++ 'sz=640x480&iu=/19968336/prebid_video_adunit&impl=s&gdfp_req=1'
++ '&env=vp&output=xml_vast2&correlator=' + Date.now();
+
+var videoUrl = pbjs.adServers.dfp.buildVideoUrl({
+    adUnit: videoAdUnit,
+    url: adserverTag
+});
+```
+
 </div>
+
+<a name="buildVideoUrl-options-object" />
+
+#### The `options` object
+
+The `options` object has the following fields:
+
+{: .table .table-bordered .table-striped }
+| Field  | Type   | Description                                                                                                                                                                         |
+|--------+--------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| adUnit | object | (Required) The Prebid adUnit to which the returned URL will map.                                                                                                                    |
+| bid    | object | (Optional) The Prebid bid for which targeting will be set. If this is not defined, Prebid will use the bid with the highest CPM for the adUnit.                                     |
+| params | object | (Required) Querystring parameters that will be used to construct the DFP video ad tag URL.  Publisher-supplied values will override values set by Prebid.js.  See below for fields. |
```

Comment:
I don't think that Options.params is required.  It should be acceptable to invoke buildVideoUrl with an Options object that contains either the "url" or "params" field.

@matthewlane can we confirm the expected behavior if both `Options.params` and `Options.url` are passed to `buildVideoUrl`?

Reply:
Right, `options.params` isn't required, the pub can pass either `options.params`, `options.url`, or both, but not neither.

If both `params` and `url` are passed in, the url is parsed, and any equivalent `params` that are present will overwrite those that were present in the url. Then everything else that happens when just params were passed in still happens (adding defaults, adserver targeting, etc.), everything is stringified back to a url, and this is returned